### DISCUSSION TITLE
How it is possible to access a selected row data?

### DISCUSSION THREAD

**Discussion by GianlucaLostuzzo** [Budibase/budibase#7582] [Help] [answered]
Hi everyone,
I'm building an app where the users can request the subscription to a learning course and my app is based on PostgreSQL DB so I have the users table, courses table and subscription table. I displayed the active courses in a table where the user can select a row to request via a button the subscription to that selected course.
My question is: is it possible to access the specific data of that selected row in order to pass the course_id to the query that I want to run in order to create the request in the DB (something like {{ Table.Selected rows[0].course_id }}) ? 
Thanks.

**Comment by melohagan** [comment]
Hi @GianlucaLostuzzo 

Yes this should be possible. There's docs on how to pass a binding value to a custom query here: https://docs.budibase.com/docs/custom-queries#using-query-bindings

**Reply by GianlucaLostuzzo** [reply]
Hi @melohagan,
I've already read that documentation and used it to set up correctly the bindings in my queries. I'm facing a problem regarding how to access the data of a selected row in a table with the bindings, in order to pass the right values to the query that I have to run.

**Reply by GianlucaLostuzzo** [reply]
<img width="1440" alt="Schermata 2022-09-05 alle 09 34 48" src="https://user-images.githubusercontent.com/64728640/188393986-4b4fdcb7-9c21-4620-b048-f96e8644e48d.png">

As you can see in this image, I would like to know how to access the "IDCORSO" value of the selected row. And how to access the value 2 of the column "IDUTENTE" in the second table. The paragraphs that you see are just a test of various bindings combination to see if the binding works.

**Comment by melohagan** [comment]
Apologies @GianlucaLostuzzo I think I misread the question! Unfortunately this is not possible for custom queries; there's an open issue here: https://github.com/Budibase/budibase/issues/7348

**Reply by GianlucaLostuzzo** [reply]
No problem @melohagan, probably I've not been so clear myself in explaining the problem so I'll try better now! 
The table showed in the image above is actually from an SQL Table and not a custom query (hence the checkbox on the left of each row enabled by the 'Allow row selection' button in the table settings).

In this screen I would like to:

- select a row
- click on the button 'Iscriviti'
- the button triggers a custom query
- the custom query needs the value present in the column 'idcorso' of the selected.

So, is it possible to access the value contained in the column 'idcorso' with the bindings? (something like {{ Table.selectedRow.idcorso }})

**Reply by JayP718** [reply]
Hi @GianlucaLostuzzo  @mike12345567 showed how this can be done in a separate thread. If I'm understanding the problem correctly, you want to select some rows in one table and use that data to filter a custom query.

If that's the case you'll need to set up the custom query to use bindings and have a where statement like Where in ({{selected_rows}}) from there you can use @mike12345567's code to grab the selected ids from the table and pass that to custom query. You might have to add a bit of code to encapsulate each element in single quotes. Hope this helps.

return $("New Table.Selected rows").map(id => JSON.parse(decodeURIComponent(id.replace(/'/g, "\"")))[0])

**Reply by GianlucaLostuzzo** [reply]
Thanks @JayP718 for the suggestion and I'll definitely keep it in mind if I need it in the future! I found another way to overcome the problem I was facing so thanks for the help @melohagan @JayP718 .

**Reply by sebastian-jansson** [reply]
> Thanks @JayP718 for the suggestion and I'll definitely keep it in mind if I need it in the future! I found another way to overcome the problem I was facing so thanks for the help @melohagan @JayP718 .

@GianlucaLostuzzo  What was your solution in the end?

**Reply by GianlucaLostuzzo** [reply, marked as answer]
Hi @sebastian-jansson I solved the problem by inserting a button in the table that triggers a custom query. Thanks to this button it was possible to access the row data needed in the custom query. I hope it helps.
